src/main/java/com/microsoft/graph/core/content/BatchRequestContent.java

```diff
@@ -208,13 +208,16 @@ private void writeBatchRequestStep(BatchRequestStep requestStep, JsonWriter writ
                     writer.value(rawBodyContent);
                 }
             }
-            if(headers.size() != 0 || requestBody != null) {
-                writer.name(CoreConstants.BatchRequest.HEADERS);
-                writer.beginObject();
-                for(int i = 0; i < headers.size(); i++) {
-                    writer.name(headers.name(i)).value(headers.value(i));
-                }
-                writer.endObject();
+            // If 'authorization' header is the only header, do not include headers object
+            if(!(headers.names().size() == 1 && headers.names().contains("authorization")) && (headers.size() != 0 || requestBody != null)) {
+                    writer.name(CoreConstants.BatchRequest.HEADERS);
+                    writer.beginObject();
+                    for (int i = 0; i < headers.size(); i++) {
+                        // If multiple headers exist, do not include 'authorization' header
+                        if(!headers.name(i).equals("authorization"))
```

Comment:
Just to confirm, do we need to normalize the casing here incase the header is of a different case?